File: python/p5.py
# Solution for Project Euler problem 5
# http://projecteuler.net/index.php?section=problems&id=5

q = 20
while 1:
    all = True
    for i in xrange(1,21):
        if q % i != 0:
            all = False
            break
    if all:
        break
    q += 20

print q

File: scala/p5.scala
// Solution for Project Euler problem 5
// http://projecteuler.net/index.php?section=problems&id=5

def gcd(a: Long, b: Long): Long = b match {
  case 0 => a
  case _ => gcd(b, a % b)
}

println((1 to 20).map(a => a.toLong).reduceLeft((a, b) => a*b/gcd(a, b)))
